### PR TITLE
Replace var with let/const, remove dead code, replace bizarre variable names with self-documenting ones

### LINKED ISSUE #13: Replace 'var' with 'let' and 'const'.
Creating this issue per prior discussion with @ryanseys; Lune is currently built to work with Node 0.10, but this restricts us to using ugly old `var`. For reasons of maintenance sanity, let's bring Lune up to date to work with the current LTS and Stable versions of Node, and make use of `let` and `const` while we're at it.

While I'm at it, any problems with my bringing in JSHint as a dev dependency and enabling some basic style checks?

### PR DESCRIPTION
**Pull request by ironwallaby**
*   Test on Node `4.4.0` and `5.8.0` instead of `0.10.0`. *[enhancement]*
*   All instances of `var` have been replaced with `let` and `const` (which is preferred whenever possible). *[style]*
*   Branches that could never, ever be reached were removed from the code entirely. *[style]*
*   Replaced confusing branch conditionals with simple `switch` statements. *[style]*
*   Replaced bizarre variable names (like `MM` or `mEc`) with sane ones (like `moon_mean_anomaly`, etc.) *[style]*
*   Tried to be more careful with types, hopefully removing all instances of using a `Date` as a number, and using `date.getTime()` instead. *[style]*

With all these single quotes, the code feels like it's speaking in a British accent :)

This PR fixes #13.
This PR includes #14, not sure how Git appreciates that.

### CODE REVIEW

File: lib/lune.js [hunk 1]
```diff
@@ -8,309 +8,241 @@
...
+  const mprime = 306.0253 + 385.81691806 * k + (0.0107306 + 0.00001236 * t) * t * t
 
-    if ((tphase < 0.01) || (Math.abs(tphase - 0.5) < 0.01)) {
+  // Moon's argument of latitude
+  const f = 21.2964 + 390.67050646 * k - (0.0016528 - 0.00000239 * t) * t * t
 
-      // Corrections for New and Full Moon
-      pt = pt + (
-        (0.1734 - 0.000393 * t) * dsin(m) +
+  switch (tphase & PHASE_MASK) {
```
Comment (ryanseys) [inline lib/lune.js]: maybe a comment explaining these switch statements.

Comment (ironwallaby) [inline lib/lune.js]: :+1:

**Comment by ryanseys** [comment]
Woo!! Shall I publish as 0.3.0?

**Comment by ironwallaby** [comment]
:+1:

**Comment by ryanseys** [comment]
Done!

**Pull request merged**